src/visitors/global_var_visitor.cpp

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <iostream>
+#include <memory>
+
+#include "ast/global.hpp"
+#include "ast/program.hpp"
+#include "ast/range.hpp"
+#include "ast/statement_block.hpp"
+#include "visitors/global_var_visitor.hpp"
+/**
+ * \file
+ * \brief AST Visitor to change ast::GlobalVar to ast::Range
+ * variables in nmodl::symtab to nmodl::symtab::syminfo::NmodlType::range_var
+ * so that they can be handled by Codegen as RANGE variables
+ */
+
+namespace nmodl {
+namespace visitor {
+
+void GlobalToRangeVisitor::visit_statement_block(ast::StatementBlock& node) {
+    const auto& statements = node.get_statements();
+    ast::RangeVarVector range_variables;
+    std::vector<std::string> global_vars_to_remove;
+    for (const auto& statement: statements) {
```

Comment:
`const` is good but there are some unnecessary ones here? `const vector` provides access to elements as if they were const.

by the way, all getters now return `const &`.

Reply:
Maybe there are some unnecessary ones. I will check again to make sure 👍